sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -771,12 +774,24 @@ def dbscan_clustering(self, cut_distance, min_cluster_size=5):
         Returns
         -------
         labels : ndarray of shape (n_samples,)
-            An array of cluster labels, one per datapoint. Unclustered points
-            are assigned the label -1.
+            An array of cluster labels, one per datapoint.
+            Outliers are labeled as follows:
+            - Noisy samples are given the label -1.
+            - Samples with infinite elements (+/- np.inf) are given the label -2.
+            - Samples with missing data are given the label -3, even if they
+              also have infinite elements.
         """
-        return labelling_at_cut(
+        labels = labelling_at_cut(
             self._single_linkage_tree_, cut_distance, min_cluster_size
         )
+        # Infer indices from labels generated during `fit`
+        infinite_index = self.labels_ == _OUTLIER_ENCODING["infinite"]["label"]
+        missing_index = self.labels_ == _OUTLIER_ENCODING["missing"]["label"]
```

Comment:
Naive question: why don't we want to alter `self.labels_`?

Reply:
`self.labels_` contains the labels generated during `fit` however this method is meant to generate labels corresponding arbitrary "cuts" using the DBSCAN method. These cuts each generate labels which may or may not align with the canonical HDBSCAN labels generated during fit. Hope this helps, let me know if there is still confusion :)